apps/glusterfs/app.go

```diff
@@ -291,6 +291,11 @@ func (a *App) SetRoutes(router *mux.Router) error {
 			Method:      "POST",
 			Pattern:     "/devices/{id:[A-Fa-f0-9]+}/state",
 			HandlerFunc: a.DeviceSetState},
+		rest.Route{
```

Comment:
@lpabon , @MohamedAshiqrh and I discussed this.

We liked that usage of api.* to indicate what to be done on the node/device instead of using another handler and REST URL. However, I don't feel "removed" fits in the hierarchy of states  "online" and "offline". Online and offline are mutually exclusive states and adding "removed" to the same hierarchy would corrupt it. A device would then could be in both "online" and "removed" state. Also, we feel that "failed" state is different than "removed". In remove we are just ensuring that all bricks from the device are moved away. It would be immediately available for new volumes.

In summary, we think, remove is an operation and not a change of state. Let us know what you think.